src/coreclr/src/vm/eventpipe.cpp

```diff
@@ -383,11 +395,15 @@ bool EventPipe::EnableInternal(
     // Enable tracing.
     s_config.Enable(*pSession, pEventPipeProviderCallbackDataQueue);
 
-    // Enable the sample profiler
-    if (enableSampleProfiler)
+    // Enable the sample profiler or defer until we can create threads
+    if (enableSampleProfiler && s_CanStartThreads)
     {
         SampleProfiler::Enable();
     }
+    else if (enableSampleProfiler && !s_CanStartThreads)
```

Comment:
I don't think you need this `!s_CanStartThreads` here? If the if statement above is false and `enableSampleProfiler` is `true` then `s_CanStartThreads` has to be false.

Reply:
Its not needed for correctness but stylistically I'd suggest the code is easier to understand if I don't have to first mentally evaluate !(A && B) && A => !B